Exporter/Exporter.php

```diff
@@ -31,35 +71,39 @@ class Exporter
      */
     public function getResponse($format, $filename, SourceIteratorInterface $source)
     {
-        switch ($format) {
-            case 'xls':
-                $writer = new XlsWriter('php://output');
-                $contentType = 'application/vnd.ms-excel';
-                break;
-            case 'xml':
-                $writer = new XmlWriter('php://output');
-                $contentType = 'text/xml';
-                break;
-            case 'json':
-                $writer = new JsonWriter('php://output');
-                $contentType = 'application/json';
-                break;
-            case 'csv':
-                $writer = new CsvWriter('php://output', ',', '"', '', true, true);
-                $contentType = 'text/csv';
-                break;
-            default:
-                throw new \RuntimeException('Invalid format');
+        if (!array_key_exists($format, $this->writers)) {
+            throw new \RuntimeException(sprintf(
+                'Invalid "%s" format, supported formats are : "%s"',
+                $format,
+                implode(', ', array_keys($this->writers))
+            ));
         }
+        $writer = $this->writers[$format];
 
         $callback = function () use ($source, $writer) {
             $handler = \Exporter\Handler::create($source, $writer);
             $handler->export();
         };
 
-        return new StreamedResponse($callback, 200, array(
-            'Content-Type' => $contentType,
+        $headers = array(
             'Content-Disposition' => sprintf('attachment; filename="%s"', $filename),
-        ));
+        );
+
+        /*
+         * NEXT MAJOR: rely on https://github.com/sonata-project/exporter/pull/90
```

Comment:
Your PR is for a new minor version of exporter. Changing version constraint for a new minor is not BC break.